base/associative.jl

```diff
@@ -252,7 +252,7 @@ const hasha_seed = UInt === UInt64 ? 0x6d35bb51952d5539 : 0x952d5539
 function hash(a::Associative, h::UInt)
     h = hash(hasha_seed, h)
     for (k,v) in a
-        h $= hash(k, hash(v))
+        h = h ⊻ hash(k, hash(v))
```

Comment:
OK, sure, thanks, but is this something for #15964, or is that about a more general approach?

Should I also deprecate `$=` there then, to really free `$` for future syntactical purposes?